### ISSUE TITLE
How to construct a new network from a model loaded?

### ISSUE THREAD

**Issue by dushoufu** [apache/incubator-mxnet#3284]
I have a pre-trained mxnet model and I can also load it.  Now i want to change the network, for example, inserting a new layer to the  existed network.  How can I do?  Preferably, I want to know how to get the symbol of every layer (such as a symbol of convolution layer) from a loaded model. I think this can be done definitely in mxnet, but I don't know.

**Comment by vchuravy** [comment]
I recently had the same issue: https://github.com/dmlc/mxnet/issues/3123

**Comment by VoVAllen** [comment]
I think you need a new model and bind all the trianed params into it.

**Comment by dushoufu** [comment]
I write my code according to the indication of #3123, but I am not sure whether it is correct.

model_args = {}
assert save_model_prefix is not None
model = mx.model.FeedForward.load(load_model_prefix, load_epoch)

internals = model.symbol.get_internals()
print internals.list_outputs()

dr7_symbol = internals["drop7_output"]

#below is the layer I add
fc8 = mx.symbol.FullyConnected(data=dr7_symbol, num_hidden=256)

fc9 = mx.symbol.FullyConnected(data=fc8, num_hidden=12000)

bn9 = mx.symbol.BatchNorm(data=fc9)

softmax = mx.symbol.SoftmaxOutput(data=bn9, name='softmax')

feature_extractor = mx.model.FeedForward(ctx=mx.gpu(), symbol=softmax,
                                         arg_params=model.arg_params,
                                         aux_params=model.aux_params,
                                         allow_extra_params=True)

**Comment by dushoufu** [comment]
By the way I also need to initialize the weights and bias of fc8 before use them

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
